src/java/org/apache/cassandra/config/DatabaseDescriptor.java

```diff
@@ -1424,9 +1425,20 @@ private static void validateSSTableFormatFactories(Iterable<SSTableFormat.Factor
         if (selectedFormat == null)
             throw new ConfigurationException(String.format("Selected sstable format '%s' is not available.", selectedFormatName));
 
+        validateWriteFormatVsStorageCompatibilityMode(selectedFormat, getStorageCompatibilityMode());
+
         return selectedFormat;
     }
 
+    @VisibleForTesting
+    public static void validateWriteFormatVsStorageCompatibilityMode(SSTableFormat<?, ?> selectedFormat, StorageCompatibilityMode mode)
+    {
+        if (selectedFormat.name().equals(BtiFormat.NAME) && mode != StorageCompatibilityMode.NONE)
+            throw new ConfigurationException(String.format("Selected sstable format '%s' is not available when in storage compatibility mode '%s'.",
```

Comment:
Perhaps we could take a guardrail-ish approach and delegate the formatting of the message to `StorageCompatibilityMode`. Something like: https://github.com/adelapena/cassandra/commit/30119694b64beb09d354984738ebc74fb447f7a9

That way we can make sure that all compatibility validation exceptions have a similar formatting. wdyt?

Reply:
I like the idea of moving the method to `StorageCompatibilityMode`. That creates coupling between `StorageCompatibilityMode` and the sstable format, but that coupling allows to clearly see and centralize all dependencies. It also nicely hides the fact that `DatabaseDescriptor. validateWriteFormatVsStorageCompatibilityMode` was a separate method just to allow unit testing. The point of using a generic method on `StorageCompatibilityMode` was guaranteeing that all similar messages have the same format, like ending in `is not available when in storage compatibility mode '%s'`. We can add back that generic message formatting when we have more than one caller, adding it now is probably overkill.